www/nginx/src/opnsense/service/templates/OPNsense/Nginx/http.conf

```diff
@@ -78,6 +78,15 @@ include opnsense_http_vhost_plugins/*.conf;
 {%   if OPNsense.Nginx.general.enabled is defined and OPNsense.Nginx.general.enabled == '1' %}
 {% for server in helpers.toList('OPNsense.Nginx.http_server') %}
 {%   set single_servername = server.servername.split(",")[0] %}
+{%   if server.naxsi_whitelist_srcip is defined and server.naxsi_whitelist_srcip != '' %}
+geo $naxsi_wlsrcip {
+    #ranges;
```

Comment:
We do have a field for trusted proxies: https://github.com/opnsense/plugins/blob/master/www/nginx/src/opnsense/mvc/app/models/OPNsense/Nginx/Nginx.xml#L564

Without it, nginx will not read the x-forwarded-for header.

add it as "proxy ...;"

since your change is not going to support ranges (will not work with this regular expresson), remove this line.

Reply:
I'm not sure if I got your point here.. but I'm testing without trusted proxies options (and the SecRules are bypassed as I expected):

![Screen Shot 2019-04-23 at 13 46 38](https://user-images.githubusercontent.com/917729/56600284-b4a6a800-65ce-11e9-81c2-50be4db46d91.png)

Reply:
trusted proxies are proxy servers, which are allowed to override the source IP. If you don't use them as well, you may always resolve the state for the IP of the previous load balancer.

Reply:
looks like this is already done by the real ip module:

https://nginx.org/en/docs/http/ngx_http_realip_module.html